### PR TITLE
refactor(call-home): use PathBuf instead of String for paths

### PR DESCRIPTION
**Pull request by niladrih**
Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

This PR uses the PathBuf type instead of the String type for filepaths. It is idiomatic to use PathBuf for paths.

### CODE REVIEW

File: call-home/src/common/constants.rs [hunk 1]
```diff
@@ -8,20 +12,42 @@ pub(crate) const PRODUCT: &str = "Mayastor";
...
+    const KEY: &str = "KEY_FILEPATH";
+    match env::var(KEY) {
+        Ok(input) => {
+            let path = Path::new(&input);
+            // Validate path.
+            match path.exists() && path.is_file() {
+                true => path.to_path_buf(),
+                false => {
+                    error!("validation failed for {} value \"{}\", falling back to default value: path must exist and must be that of a file", KEY, &input);
+                    Path::new(DEFAULT_ENCRYPTION_KEY_FILEPATH).to_path_buf()
```
Comment (tiagolobocastro) [inline call-home/src/common/constants.rs:46]: Is there any point in falling back? seems here we could simply panic

**Review by tiagolobocastro — APPROVED** [review]

**Comment by niladrih** [comment]
bors try

**Comment by niladrih** [comment]
bors merge

**Review by datacore-vkumar — APPROVED** [review]
LGTM

**Comment by niladrih** [comment]
bors merge

**Pull request merged**